src/megatron/bridge/data/vlm_datasets/mock_provider.py

```diff
@@ -0,0 +1,256 @@
+            "attention_mask": attention_mask,
+            "loss_mask": loss_mask,
+            "position_ids": position_ids,
+        }
+        if pixel_values is not None:
+            sample["pixel_values"] = pixel_values
+        if image_grid_thw is not None:
+            sample["image_grid_thw"] = image_grid_thw
+
+        return sample
+
+
+@dataclass(kw_only=True)
+class MockVLMConversationProvider(DatasetProvider):
+    """DatasetProvider for generic mock VLM conversation datasets.
+
+    Builds train/valid/test datasets using a HF AutoProcessor and the
+    `MockVLMConversationDataset` implementation. Intended to work across
+    different VLM models whose processors support the conversation schema.
+    """
+
+    # Required to match model.seq_length
+    sequence_length: int
+
+    # HF processor/model ID (e.g., Qwen/Qwen2.5-VL-3B-Instruct or other VLMs)
+    hf_processor_path: str
+
+    # Sample generation options
+    prompt: str = "Describe this image."
+    random_seed: int = 0
+    image_size: Tuple[int, int] = (256, 256)
+    pad_to_max_length: bool = True
+    create_attention_mask: bool = True
+
+    # Keep parity with GPTDatasetConfig usage in batching utilities
+    skip_getting_attention_mask_from_dataset: bool = True
+
+    # Number of images per sample
+    num_images: int = 1
+
+    # HF AutoProcessor instance will be set during build
+    _processor: Optional[Any] = None
+
+    def _make_base_examples(self) -> List[Dict[str, Any]]:
+        # Single minimal conversation example; dataset will repeat to target length
+        num_images = max(0, int(getattr(self, "num_images", 1)))
+        w, h = self.image_size
+        rng = numpy.random.default_rng(seed=self.random_seed)
+        images = None
+        if num_images > 0:
+            # Embed in-memory PIL images directly in the conversation so that
+            # qwen_vl_utils.process_vision_info can discover them.
+            images = [
+                Image.fromarray(rng.integers(low=0, high=256, size=(h, w, 3), dtype=numpy.uint8), mode="RGB")
+                for _ in range(num_images)
+            ]
+
+        content = [{"type": "image", "image": img} for img in images] if images is not None else []
+        content.append({"type": "text", "text": self.prompt})
+        messages = [
+            {"role": "user", "content": content},
+            {"role": "assistant", "content": "dummy assistant response"},
+        ]
+        return [{"conversation": messages}]
+
+    def build_datasets(self, context: DatasetBuildContext):
+        from transformers import AutoProcessor
+
+        # Initialize and store processor
+        self._processor = AutoProcessor.from_pretrained(self.hf_processor_path, trust_remote_code=True)
+
+        base_examples = self._make_base_examples()
+
+        def _maybe_make(size: int) -> Optional[VLMConversationDataset]:
+            if not size or size <= 0:
+                return None
+            return VLMConversationDataset(
```

Comment:
is this intentional ? shouldn't it invoke `MockVLMConversationDataset ` in stead ?